site/gatsby-site/src/components/forms/SubmissionWizard/StepOne.js

```diff
@@ -130,22 +135,33 @@ const FormDetails = ({
     resetForm,
   } = useFormikContext();
 
+  const saveInLocalStorage = debounce((values) => {
```

Comment:
I'm saying that it is probably not doing what you describe because a new function is created on every render. You need to do something like this: https://github.com/responsible-ai-collaborative/aiid/blob/a935caccf588c83f8c7b6cce07f7aaa78a6df5ed/site/gatsby-site/src/components/taxa/TaxonomyForm.js#L36